### PR TITLE
CIFuzz: switch to systemd-detect-virt to detect docker

### PR DESCRIPTION
**Pull request by evverx**
Closes https://github.com/google/oss-fuzz/issues/4093

### CODE REVIEW

File: infra/utils.py [hunk 1]
```diff
@@ -89,14 +89,11 @@ def get_fuzz_targets(path):
...
   """Gets the name of the current docker container you are in.
-  /proc/self/cgroup can be used to check control groups e.g. Docker.
-  See: https://docs.docker.com/config/containers/runmetrics/ for more info.
 
   Returns:
     Container name or None if not in a container.
   """
-  with open('/proc/self/cgroup') as file_handle:
-    if 'docker' not in file_handle.read():
+  if b'docker' not in subprocess.run('systemd-detect-virt -c', shell=True, stdout=subprocess.PIPE).stdout:
```
Comment (inferno-chromium) [inline infra/utils.py:96]: Thank you so much! you rock!

can you please do
result = subprocess.run('systemd-detect-virt -c', shell=True, stdout=subprocess.PIPE)
if b'docker' in result.stdout:

Comment (inferno-chromium) [reply infra/utils.py:96]: stdout=subprocess.PIPE can be skipped as well i think

Comment (jonathanmetzman) [reply infra/utils.py:96]: `subprocess.run(['systemd-detect-virt', '-c']).stdout` should work.

Comment (evverx) [reply infra/utils.py:96]: According to https://docs.python.org/3/library/subprocess.html#subprocess.run, `run` doesn't capture stdout by default. Another option would be to pass `capture_output=True` to it but it was added relatively recently so I don't expect it to be available on Ubuntu (where python-3.5 is still used).

Comment (inferno-chromium) [reply infra/utils.py:96]: Why not just subprocess.check_output
This is equivalent to:
run(..., check=True, stdout=PIPE).stdout

Comment (evverx) [reply infra/utils.py:96]: `systemd-detect-virt -c` returns 1 when it isn't run in containers, which makes `check_output` throws exceptions left and right. I don't think it's desirable given that that code can be run elsewhere.

Comment (inferno-chromium) [reply infra/utils.py]: ok then, please move this to a local var, that line is too long. rest lgtm.

Comment (evverx) [reply infra/utils.py]: I replaced `shell=True` with `['systemd-detect-virt', '-c']` as @jonathanmetzman suggested.

I'll add `result=...` shortly

**Comment by inferno-chromium** [comment]
Minor lint failure in presubmit in https://github.com/google/oss-fuzz/pull/4101/checks?check_run_id=855435834
```
-  result = subprocess.run(['systemd-detect-virt', '-c'], stdout=subprocess.PIPE).stdout
+  result = subprocess.run(['systemd-detect-virt', '-c'],
+                          stdout=subprocess.PIPE).stdout
```

**Comment by evverx** [comment]
The presubmit check is still failing unfortunately with
```
************* Module utils
infra/utils.py:96:11: W1510: Using subprocess.run without explicitly set `check` is not recommended. (subprocess-run-check)

-----------------------------------
Your code has been rated at 9.81/10
```

**Comment by inferno-chromium** [comment]
> The presubmit check is still failing unfortunately with
> 
> ```
> ************* Module utils
> infra/utils.py:96:11: W1510: Using subprocess.run without explicitly set `check` is not recommended. (subprocess-run-check)
> 
> -----------------------------------
> Your code has been rated at 9.81/10
> ```

it is ok, we will fix it sometime later.

**Comment by evverx** [comment]
> it is ok, we will fix it sometime later

Until then it would probably make sense to ignore subprocess-run-check there. I've just opened https://github.com/google/oss-fuzz/pull/4102.

**Pull request merged**